datafusion/physical-expr/src/window/lead_lag.rs

```diff
@@ -238,9 +238,8 @@ fn get_default_value(
 ) -> Result<ScalarValue> {
     if let Some(default_value) = default_value {
         let default_value_type = default_value.data_type();
-        if can_coerce_from(dtype, &default_value_type)
-            || (dtype.is_integer() && default_value_type.is_integer())
-        {
+
+        if comparison_coercion(&default_value_type, dtype).is_some() {
```

Comment:
It seems that you just need to change integer, since I'm not sure having `pub fn comparison_binary_numeric_coercion` is a good idea or not. At least, we can add the comment for now.

Reply:
Sorry, I didn't understand what you mean by "change integer". Could you rephrase, or provide a suggestion for a change?

Reply:
Sorry, I mean coerce, not change.

Reply:
```rust
if can_coerce_from(dtype, &default_value_type)
            || (dtype.is_integer() && default_value_type.is_integer())
```
// We only check coercion between integer.

Reply:
So just to be clear, are you suggesting that the comparison should be reverted to 
```rust
if can_coerce_from(dtype, &default_value_type)
            // We explicitely allow a lossy conversion between signed and unsigned integers
            || (dtype.is_integer() && default_value_type.is_integer())
```
?

Reply:
I was suggesting `comparison_coercion` with comment.

But, I rethink about it again, now I think  
```rust
if can_coerce_from(dtype, &default_value_type)
            || (dtype.is_integer() && default_value_type.is_integer())
```
is better for here.

btw, i dont think your comment `// We explicitely allow a lossy conversion between signed and unsigned integers` is correct 🤔

Reply:
I thought you were just trying to deal with integer type. But, I think we need to cover any compatible type 🤔 ?

If that is true, we can stick with `if comparison_coercion(&default_value_type, dtype).is_some()`

Reply:
I suggest `comparison_coercion`. I read the doc, it does not seem lag/lead should only be integer.

Reply:
cc @alamb to confirm my suggestion.

Reply:
I think coercing during physical evaluation is too late in the process -- most other coercion happens in the analyze phase so that subsequent transformations / optimizations use the same rules.  

Would it be possible to move this coercion earlier into 

https://github.com/apache/arrow-datafusion/blob/513fd052bdbf5c7a73de544a876961f780b90a92/datafusion/optimizer/src/analyzer/type_coercion.rs#L64-L63

?